app/utils/validators.py

```diff
@@ -10,10 +14,20 @@ def password_validator(password: str) -> str:
     This function is intended to be used as a Pydantic validator:
     https://pydantic-docs.helpmanual.io/usage/validators/#reuse-validators
     """
-    # TODO
-    if len(password) < 6:
-        raise ValueError("The password must be at least 6 characters long")
-    return password.strip()
+    password = password.strip()
+    result = zxcvbn.zxcvbn(password)
+    if not result["score"] == 4:
+        raise ValueError(
+            result["feedback"],
+        )
+    if not re.fullmatch(
+        r"(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$€%^&*\(\)_+\-.,.?\":\{\}|<>'/;\[\]]).{6,}",
+        password,
+    ):
+        raise ValueError(
+            "The password must be at least 6 characters long and contain at least one number, one special character, one majuscule and one minuscule.",
+        )
```

Comment:
We can remove the regex
```suggestion
```

Reply:
No we can't as "azdgnuhmokbh" would be a valid password because it has a score of 4 for zxcvbn